File: recursive_sort.rb
#recursively sort an array

def quick_sort(array)
  if array == []
    []
  else
    pivot = array[0]
    left, right = array.drop(1).partition {|x| x <= pivot }
    quick_sort(left) + [pivot] + quick_sort(right)
  end
end

p quick_sort([9,3,7,8,5,4,3,2,1])
p quick_sort([])

File: recursive_sort.js
var recur_sort = function(array){
  if (array.length === 0){
    return [];
  } else {
    var left = [];
    var right = [];
    var pivot = array[0]
    array.slice(1).forEach(function(x){
      if (x <= pivot){
        left.push(x)
      } else {
        right.push(x)
      }
    })
   return [].concat(recur_sort(left), [pivot], recur_sort(right))
   }
}

console.log(recur_sort([4,7,9,5,3,2,8]))
